### PR TITLE
chore: upgrade to Erlang 24

### PR DESCRIPTION
**Pull request by boringcactus**
Erlang/OTP 22 doesn't build on GitHub Actions `ubuntu-latest`, because OpenSSL 3 compatibility was only introduced in Erlang/OTP 24.2.

Elixir 1.11.4 is the oldest version that's declared to be compatible with OTP 24.

All of these library version upgrades are the smallest version bump required to build without deprecation warnings.

### CODE REVIEW

**Review by paulswartz — COMMENTED** [review]
question: Is there a reason we can't switch the GitHub Actions runner to `ubuntu-20.04` and keep support for older version of Erlang/Elixir?

**Comment by boringcactus** [comment]
I guess we could do that in the short term if we decided that we wanted to, but the Ubuntu 18.04 runner [was deprecated](https://github.com/actions/runner-images/issues/6002) once GitHub Actions added support for Ubuntu 22.04, so it seems like the Ubuntu 20.04 runner will be going away once there's a 24.04 runner, and Ubuntu 24.04 is scheduled to come out next week. If we don't do this now, we'll have to do this later, anyway.

File: mix.exs [hunk 1]
```diff
@@ -5,13 +5,13 @@ defmodule ServerSentEventStage.MixProject do
     [
       app: :server_sent_event_stage,
       version: "1.1.0",
-      elixir: "~> 1.9",
+      elixir: "~> 1.11",
```
Comment (paulswartz) [inline mix.exs]: suggestion: this isn't quite right, since `1.11.3` would match.
```suggestion
      elixir: ">= 1.11.4 and < 2.0.0",
```

**Review by paulswartz — COMMENTED** [review]
I poked around, and OpenSSL 1.x isn't supported either so I don't feel as bad dropping support for that reason. One comment about the Elixir version and then this will be good to go.

**Review by paulswartz — APPROVED** [review]

**Pull request merged**